src/execute_linter.cc

```diff
@@ -24,24 +25,172 @@
 #include "absl/strings/string_view.h"
 #include "src/lint_errors.h"
 #include "src/linter.h"
+#include "src/linter_options.h"
 #include "zetasql/base/status.h"
 #include "zetasql/base/status_macros.h"
 
 namespace zetasql::linter {
 
 namespace {
+
+// This function will parse "NOLINT"(<CheckName>) syntax and adjust enability
+// options for each check. If NOLINT usage errors occur, it counts as a lint
+// error and it will be returned in a result.
+// This works similar to consistent comment type check.
+LinterResult ParseNoLintComments(absl::string_view sql, LinterOptions options,
+                                 std::map<ErrorCode, LinterOptions>* nolint) {
+  LinterResult result;
+  for (int i = 0; i < static_cast<int>(sql.size()); ++i) {
```

Comment:
I think I saw this code in some other check. Can you somehow extract iterating over the input ignoring multiline tokens into a separate function?

Reply:
It is the same as comments check. I mentioned it explanation. It seemed to me that it is more bothersome to extract "ignore inside string" and "ignore multiline comments" parts, considering they are in seperate files, has different inputs and are only like 20-30 lines.

Reply:
well, I disagree :) If you have the same code in multiple places, then someone for sure will forget updating all places when changing something in one place. This way you get similar bugs popping up again and again in different parts of the application. Please, create a new issue to reduce code duplication when iterating over characters. No need to fix it in the current pr.

Reply:
+1 for Andrey comment. Can you please a TODO for extracting this part of the code in a helper function?